### DISCUSSION TITLE
Github Actions Python subprocess.run

### DISCUSSION THREAD

**Discussion by codebykyle** [community/community#26197] [Actions]
<p>Hello,<br>
I have a python file which I’d like to use as part of a Github Action. It uses subprocess.run(). For some reason, this fails in a Github Action.</p>
<p>Here is the code in question, where “arg_method” is the string “decrypt”:</p>
<pre><code>for filein files_to_handle:  
process= subprocess.run(  
 [  
"gcloud",  
"kms",  
arg_method,  
"--location=" + hidden,  
"--keyring=" + hidden,  
"--key=" + hidden,  
"--plaintext-file=" + hidden,  
"--ciphertext-file=" + hidden  
 ],  
shell=True,  
check=True  
)
</code></pre>
<p>The github action always fails on this step. Here is the output:</p>
<pre><code>Run python python_script_name_hidden.py --method=decrypt --env=production --yes
ERROR: (gcloud) Command name argument expected.
</code></pre>
<p>It appears the arguments are not being passed to the subprocess. It is not getting as far as “kms”</p>
<p>Any advice on how to fix this?</p>
<p>Thank you.</p>

**Comment by codebykyle** [comment, marked as answer]
<p>Hey all. I solved this. Running this on a Linux server revealed it wasn’t working everywhere.</p>
<p>I changed the script to the following, which seems to have worked:</p>
<pre><code>for file in files_to_handle:
    process = subprocess.Popen(
        [
            "gcloud",
            "kms",
            arg_method,
            "--location=" + hidden,
            "--keyring=" + hidden,
            "--key=" + hidden,
            "--plaintext-file=" + hidden,
            "--ciphertext-file=" + hidden
        ],
        stdout=subprocess.PIPE,
        stderr=subprocess.PIPE
    )

    output, error = process.communicate()

    if process.returncode != 0:
        raise Exception("File handling failed %d %s %s" % (process.returncode, output, error))
</code></pre>
<p>I thought I’d update this incase anyone has a similar issue</p>

**Reply by stardustman** [reply]
I'm curious about why subprocess.run not work?

**Reply by JonZeolla** [reply]
As am I 👀

**Comment by BrightRan** [comment]
<p><a class="mention" href="/codebykyle">@codebykyle</a> ,</p>
<p>Glad to hear that the problem has been solved. And thanks for sharing your experiences.</p>

**Comment by codebykyle** [comment]
<p>Hey <a class="mention" href="/brightran">@brightran</a> </p>
<p>Thanks for the quick reply. I ran this on another machine and was able to replicate the issue. I solved this by using subprocess.Popen as opposed to subprocess.run.</p>
<p>Thanks for the suggestions. I marked the previous reply as a solution to avoid any confusion.</p>
<p>Thanks,</p>
<p>-Kyle</p>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/codebykyle">@codebykyle</a> ,</p>
<p>Yeah, I didn’t noticed your previous reply until I submitted my reply, so I changed my reply.<br>
Anyway, thanks again for share your experiences here. I believe this can help more people who have the same/similar problems.</p>
